File: backend/internal/model/sesion.go
package model

// TipoSesion representa el tipo de sesión dentro de un Gran Premio.
type TipoSesion string

const (
	TipoQualy       TipoSesion = "qualy"
	TipoSprintQualy TipoSesion = "sprint_qualy"
	TipoSprint      TipoSesion = "sprint"
	TipoCarrera     TipoSesion = "carrera"
)

// EstadoSesion representa el estado de una sesión.
type EstadoSesion string

const (
	EstadoSesionPendiente  EstadoSesion = "pendiente"
	EstadoSesionCompletada EstadoSesion = "completada"
)

// Sesion representa una sesión (qualy, sprint o carrera) de un Gran Premio.
type Sesion struct {
	ID           int          `json:"id"`
	GranPremioID int          `json:"gran_premio_id"`
	Tipo         TipoSesion   `json:"tipo"`
	Estado       EstadoSesion `json:"estado"`
}

File: frontend/src/types/sesion.ts
export type TipoSesion = "qualy" | "sprint_qualy" | "sprint" | "carrera";
export type EstadoSesion = "pendiente" | "completada";

export interface Sesion {
  id: number;
  gran_premio_id: number;
  tipo: TipoSesion;
  estado: EstadoSesion;
}

export interface InscripcionPublica {
  piloto_id: number;
  nombre_piloto: string;
  equipo_id: number;
  nombre_equipo: string;
}

export interface ResultadoPublico {
  id: number;
  posicion: number | null;
  puntos: number | null;
  vuelta_rapida: boolean;
  inscripcion: InscripcionPublica;
}

export interface SesionConResultados {
  id: number;
  tipo: TipoSesion;
  estado: EstadoSesion;
  resultados: ResultadoPublico[];
}
